pkg/kubelet/qos/util/qos.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+)
+
+const (
+	Guaranteed = "Guaranteed"
+	Burstable  = "Burstable"
+	BestEffort = "Best-Effort"
+)
+
+// isResourceGuaranteed returns true if the container's resource requirements are Guaranteed.
+func isResourceGuaranteed(container *api.Container, resource api.ResourceName) bool {
+	// A container resource is guaranteed if its request == limit.
+	// If request == limit, the user is very confident of resource consumption.
+	req, hasReq := container.Resources.Requests[resource]
+	limit, hasLimit := container.Resources.Limits[resource]
+	if !hasReq || !hasLimit {
+		return false
+	}
+	return req.MilliValue() == limit.MilliValue() && req.MilliValue() != 0
```

Comment:
This can cause errors in the opposite direction with large numbers.

Please make this the following:

```
return req.Cmp(limit) == 0 && req.Value() != 0
```